lib/bike_brigade_web/components/sms_message_list_component.ex

```diff
@@ -0,0 +1,139 @@
+defmodule BikeBrigadeWeb.Components.SMSMessageListComponent do
+  use Phoenix.Component
+  use BikeBrigadeWeb, :html
+  alias BikeBrigade.Messaging.SmsMessage
+
+
+  def preview_message(%SmsMessage{body: body}) when not is_nil(body) do
+    if String.length(body) > 30 do
+      String.slice(body, 0, 30) <> "..."
+    else
+      body
+    end
+  end
+
+  def preview_message(%SmsMessage{media: media}) when length(media) > 0 do
+    "Attachment: #{length(media)} media"
+  end
+
+  def preview_message(_), do: "unknown message"
+
+
+  def handle_build_message_link({"/messages", rider_id}) do
+    ~p"/messages/#{rider_id}"
+  end
+
+  def handle_build_message_link({"/campaign", campaign_id, rider_id}) do
+    ~p"/campaigns/#{campaign_id}/messaging/riders/#{rider_id}"
+  end
+
+
+  # TODO: can :any be a tuple?
```

Comment:
I believe so!

Reply:
We can make a struct here for `RiderMessage` but i think that's overkill for now....

Reply:
Duly noted.